packages/dd-trace/src/log/index.js

```diff
@@ -20,13 +22,29 @@ function processMsg (msg) {
   return typeof msg === 'function' ? msg() : msg
 }
 
+const config = {
+  enabled: false,
+  logger: undefined,
+  logLevel: 'debug'
+}
+
 const log = {
+  /**
+   * @returns Read-only version of logging config. To modify config, call `log.use` and `log.toggle`
+   */
+  getConfig () {
+    return Object.freeze({ ...config })
```

Comment:
Yes it will throw. If we're concerned about crashing, we could skip the `freeze` part.

My thought was that I'd rather fail as early as possible than risk a bug being introduced because it was possible to modify the returned config (as someone might mistakingly think they could use it to modify the config).

Reply:
Yeah i totally understand your intention here, and i'd agree if it was under a different code constraints. In my opinion, i'd rather risk a config not being applied, than a customer's app crashing, it's a very important constraint we try to follow.
Now it could be interesting to test this on our own apps for a bit, see if any crash happen. But doing this in customer's prod would be basically letting them find bugs for us at a great cost for them.
I'll let you and the others be the judge of the risk reward ratio, i'm just adding some context here :+1:

Reply:
Fixed in #a99a2f3 👍